### PR TITLE
Introduce VerifyTokenHash to support the PKCE flow for email signup

### PR DESCRIPTION
**Pull request by alexbakker**
Relevant documentation:
https://supabase.com/docs/guides/auth/passwords?queryGroups=flow&flow=pkce#signing-up-with-an-email-and-password

### CODE REVIEW

File: Gotrue/Interfaces/IGotrueApi.cs
Comment (acupofjose) [inline Gotrue/Interfaces/IGotrueApi.cs:1]: This should be changed to the generic `TSession`

**Review by acupofjose — CHANGES_REQUESTED** [review]
Awesome work, thank you! Just minor change and we should be good to go.

**Comment by alexbakker** [comment]
Fixed!

**Comment by acupofjose** [comment]
LGTM! I’ll add it to the next release

**Comment by acupofjose** [comment]
Available in 5.0.6 and will be included shortly in the next Supabase minor version

**Pull request merged**